konrad/humidity/relative_humidity.py

```diff
@@ -264,159 +270,206 @@ def __call__(self, atmosphere, **kwargs):
+            freezing_pt_rh (float in [0;1]): value of relative humidity at the freezing point.
+            bl_top_p (float): Pressure of the top of the boundary layer (bl) where the humidity peak is.
+            bl_top_rh (float in [0;1]): value of the relative humidity at the top of the boundary layer.
+            surface_rh (float in [0;1]): value of the relative humidity at the surface.
         """
 
-        ## Convert percent to dimensionless 
-        if top_peak_rh > 1 : top_peak_rh /= 100;
-        if mid_rh > 1 : mid_rh /= 100;
-        if low_peak_rh > 1 : low_peak_rh /= 100;
+        ## Convert percent to dimensionless
+        if top_peak_rh > 1:
+            top_peak_rh /= 100
+        if freezing_pt_rh > 1:
+            freezing_pt_rh /= 100
+        if bl_top_rh > 1:
+            bl_top_rh /= 100
+        if surface_rh > 1:
+            surface_rh /= 100
 
         # Affect values to self
-        self.top_slope = top_slope
+        self.top_peak_T = top_peak_T
         self.top_peak_rh = top_peak_rh
-        self.mid_p = mid_p
-        self.mid_rh = mid_rh
-        self.low_peak_p = low_peak_p
-        self.low_peak_rh = low_peak_rh
-        self.bl_slope = bl_slope
+        self.freezing_pt_rh = freezing_pt_rh
+        self.bl_top_p = bl_top_p
+        self.bl_top_rh = bl_top_rh
+        self.surface_rh = surface_rh
 
     def __call__(self, atmosphere, **kwargs):
         """
-        Parameters: 
+        Parameters:
             atmosphere (konrad.atmosphere.Atmosphere): The atmosphere component.
 
         Returns:
             ndarray: The relative humidity profile.
         """
-        
+
         plev = atmosphere["plev"]
+        T = atmosphere["T"][-1, :]
+
+        ## Boundary layer
+        bl_slope = (self.bl_top_rh - self.surface_rh) / (self.bl_top_p - 1000e2)
 
-        ## Top layer
-        cold_point_p = atmosphere.get_cold_point_plev()
-        tl_func = lambda p : self.top_peak_rh/2 + self.top_slope * (p - cold_point_p)
-        top_peak_p = cold_point_p + self.top_peak_rh / (2 * self.top_slope)
+        def bl_func(p):
+            return self.bl_top_rh + bl_slope * (p - self.bl_top_p)
 
-        ## Mid-troposphere
-        coeffs = np.polyfit([top_peak_p, self.mid_p, self.low_peak_p], [self.top_peak_rh, self.mid_rh, self.low_peak_rh], deg = 2)
-        mid_func = lambda p : coeffs[0] * p ** 2 + coeffs[1] * p + coeffs[2]
+        bl_rh = bl_func(plev[plev > self.bl_top_p])
 
-        ## Boundary layer
-        bl_func = lambda p : self.low_peak_rh + self.bl_slope*(p - self.low_peak_p)
+        ## Between the top of the b.l. and the freezing point (fp)
+        fp_p = atmosphere.get_freezing_point_plev()
+        # Quadratic function of p going through both point with a zero slope at freezing level:
+        def bottom_func(p):
+            return (self.bl_top_rh - self.freezing_pt_rh) / (
+                self.bl_top_p - fp_p
+            ) ** 2 * (p - fp_p) ** 2 + self.freezing_pt_rh
```

Comment:
Did you perform simulations with a bottom function that was actually fixed in pressure? So not to the pressure of the freezing level? Currently, the `RH(p)` in the lower troposphere will change with warming which might introduce a feedback.

Reply:
I'm aware of that, and I chose this implementation because it spared me from wondering how to connect between the triple point pressure and what would be the fixed pressure...

Reply:
For the latter question, I would simply choose a pressure height in a way that makes the lower-tropospheric RH agree with ERA5. The first question is a bit more tricky (I also gave it a try and it is indeed hard to ensure a smooth transition for a wide range of temperatures). However, I am not sure if it is too important that the transition at the freezing point is smooth. It will likely affect the heating rates but I am not sure if that's relevant for a first-order test of the impact on climate sensitivity.